vault/logical_system_activity.go

```diff
@@ -186,8 +188,16 @@ func (b *SystemBackend) handleActivityConfigUpdate(ctx context.Context, req *log
 	{
 		// Parse the enabled setting
 		if enabledRaw, ok := d.GetOk("enabled"); ok {
-			config.Enabled = enabledRaw.(string)
+			enabledStr := enabledRaw.(string)
+
+			// If currently enabled with the intent to disable or intent to revert to
+			// default in a OSS context, then we return a warning to the client.
+			if config.Enabled == "enable" && enabledStr == "disable" ||
+				!activityLogEnabledDefault && config.Enabled == "enable" && enabledStr == "default" {
+				warnings = append(warnings, "the current monthly segment will be deleted because the activity log was disabled")
+			}
```

Comment:
There are three transitions where we need to warn

enabled => disabled                                                    // part before ||, OK
enabled => default && !activityLogEnabledDefault    // part after ||, OK
default && activityLogEnabledDefault => disabled    // I don't see how this gets handled

Reply:
`default && activityLogEnabledDefault => disabled` It's not handling this. I'll fix it.

Reply:
`default && activityLogEnabledDefault => disabled` It's not handling this. I'll fix it.

Reply:
`default && activityLogEnabledDefault => disabled` It's not handling this. I'll fix it.